File: ios/managing-navigation/GetArticlesSlug.swift
// DocSection: managing_navigation_articles_slugs
// Tip: Find more about Swift SDK at https://docs.kontent.ai/ios
import KenticoCloud
 
let client = DeliveryClient.init(projectId:"975bf280-fd91-488c-994c-2f04416e5ee3")
 
let customQuery = "items?system.type=article&elements=title,url_pattern"
client.getItems(modelType: Article.self, customQuery: customQuery) { (isSuccess, itemsResponse, error) in
      if isSuccess {
           if let articles = itemsResponse?.items {
               // Use your items here
           }
       } else {
           if let error = error {
               print(error)
           }
       }
// EndDocSection

File: android/managing-navigation/GetArticlesSlug.java
// DocSection: managing_navigation_articles_slugs
// Tip: Find more about JavaRx SDK at https://docs.kontent.ai/androidandroid

// Prepares the DeliveryOptions configuration object
DeliveryOptions options = DeliveryOptions.builder()
    .projectId("<YOUR_PROJECT_ID>")
    .build();

// Initializes a DeliveryClient for Java projects
DeliveryClient client = new DeliveryClient(options);

// Registers the model class for articles
client.registerType(Article.class);

// Gets specific elements of all articles using a simple request
CompletionStage<List<Article>> articles = client.getItems(
    Article.class, 
    DeliveryParameterBuilder.params().projection("title", "url_pattern").build()
);

// Gets specific elements of all articles using JavaRx
Observable.fromCompletionStage(client.getItems(
    Article.class,
    DeliveryParameterBuilder.params().projection("title", "url_pattern").build()
)).subscribe(new Observer<List<Article>>() {
    @Override
    public void onSubscribe(@NonNull Disposable d) {
    }

    @Override
    public void onNext(@NonNull List<Article> items) {
        // Gets the content items
        List<Article> articles = items;
    }

    @Override
    public void onError(@NonNull Throwable e) {
    }

    @Override
    public void onComplete() {
    }
});
// EndDocSection